patches/chromium/convert_electron_processsingleton_changes_to_patch.patch

```diff
@@ -0,0 +1,159 @@
+index 0291db0f919aa7868d345f4f0712c42c6ad7ee17..1c34b939e85543aba6801c45ba81fe626aad65fe 100644
+--- a/chrome/browser/process_singleton.h
++++ b/chrome/browser/process_singleton.h
+@@ -103,7 +103,8 @@ class ProcessSingleton {
+       base::RepeatingCallback<bool(const base::CommandLine& command_line,
+                                    const base::FilePath& current_directory)>;
+ 
+-  ProcessSingleton(const base::FilePath& user_data_dir,
++  ProcessSingleton(const std::string& program_name,
++                   const base::FilePath& user_data_dir,
+                    const NotificationCallback& notification_callback);
+   ~ProcessSingleton();
+ 
+@@ -166,6 +167,7 @@ class ProcessSingleton {
+ 
+  private:
+   NotificationCallback notification_callback_;  // Handler for notifications.
++  std::string program_name_; // Used for kMutexName.
+ 
+ #if defined(OS_WIN)
+   bool EscapeVirtualization(const base::FilePath& user_data_dir);
+diff --git a/chrome/browser/process_singleton_posix.cc b/chrome/browser/process_singleton_posix.cc
+index dc9c1b76a1c7c8b3fa83fc83788eef36d2cfa4a5..c65419336ace0cf7b4a975ebb2d2307fefbe966d 100644
+--- a/chrome/browser/process_singleton_posix.cc
++++ b/chrome/browser/process_singleton_posix.cc
+@@ -348,6 +348,21 @@ bool CheckCookie(const base::FilePath& path, const base::FilePath& cookie) {
+   return (cookie == ReadLink(path));
+ }
+ 
++bool IsAppSandboxed() {
++#if defined(OS_MAC)
++  // NB: There is no sane API for this, we have to just guess by
++  // reading tea leaves
++  base::FilePath home_dir;
++  if (!base::PathService::Get(base::DIR_HOME, &home_dir)) {
++    return false;
++  }
++
++  return home_dir.value().find("Library/Containers") != std::string::npos;
++#else
++  return false;
++#endif  // defined(OS_MAC)
++}
++
+ bool ConnectSocket(ScopedSocket* socket,
+                    const base::FilePath& socket_path,
+                    const base::FilePath& cookie_path) {
+@@ -712,9 +727,11 @@ void ProcessSingleton::LinuxWatcher::SocketReader::FinishWithACK(
+ // ProcessSingleton
+ //
+ ProcessSingleton::ProcessSingleton(
++    const std::string& program_name,
+     const base::FilePath& user_data_dir,
+     const NotificationCallback& notification_callback)
+     : notification_callback_(notification_callback),
++      program_name_(program_name),
+       current_pid_(base::GetCurrentProcId()),
+       watcher_(new LinuxWatcher(this)) {
+   socket_path_ = user_data_dir.Append(chrome::kSingletonSocketFilename);
+@@ -997,12 +1014,26 @@ bool ProcessSingleton::Create() {
+ #endif
+   }
+ 
+-  // Create the socket file somewhere in /tmp which is usually mounted as a
+-  // normal filesystem. Some network filesystems (notably AFS) are screwy and
+-  // do not support Unix domain sockets.
+-  if (!socket_dir_.CreateUniqueTempDir()) {
+-    LOG(ERROR) << "Failed to create socket directory.";
+-    return false;
++  if (IsAppSandboxed()) {
```

Comment:
Looks like this issue got addressed in https://codereview.chromium.org/2811173003, is this needed ?

Reply:
We can keep this code because it allows Electron apps on the Mac running under the Mac sandbox to try and set up a shorter `socket_dir` to use for the socket, rather than just fail to set up the socket completely due to the path being too long.